xmtp_mls/src/groups/tests/test_consent.rs

```diff
@@ -0,0 +1,32 @@
+use xmtp_db::consent_record::ConsentState;
+
+use crate::tester;
+
+#[xmtp_common::test(unwrap_try = "true")]
+async fn test_auto_consent_to_own_group() {
+    tester!(alix1);
+
+    tester!(bo);
+    let unwanted_bo = bo
+        .create_group_with_inbox_ids(&[alix1.inbox_id()], None, None)
+        .await?;
+
+    alix1.sync_welcomes().await?;
+    let unwanted = alix1.group(&unwanted_bo.group_id)?;
+    // We were added by ourselves, but we did not consent to the group. The group should remain unconsented.
+    assert_eq!(unwanted.consent_state()?, ConsentState::Unknown);
+
+    tester!(alix2, from: alix1);
+    unwanted.send_message(b"hi unwanted group").await?;
```

Comment:
I thought that sending a message in a group auto consented to the group... did I make that up? Was that a V2 only thing. 👀